cmd/rofl/mgmt.go

```diff
@@ -117,21 +117,60 @@ var (
 				cobra.CheckErr(fmt.Errorf("failed to write manifest: %w", err))
 			}
 
-			// Initialize git repository if not already present.
-			_, err = os.Stat(".git")
-			if err != nil && errors.Is(err, os.ErrNotExist) {
-				// Git directory doesn't exist, try to initialize.
-				gitInitCmd := exec.Command("git", "init")
-				if err = gitInitCmd.Run(); err != nil {
-					fmt.Printf("Git repository not initialized: %s.\n", err)
+			// Save current working directory.
+			prevWD, err := os.Getwd()
+			cobra.CheckErr(err)
+			curWD := prevWD
+
+			// Find root of git repository if present.
+			var foundGit bool
+			var curPath string
+			for !foundGit {
+				curPath = filepath.Dir(curWD)
+				if curPath == filepath.VolumeName(os.TempDir())+string(os.PathSeparator) {
+					// Root of filesystem reached, stop traversal.
+					break
+				}
+
+				_, err = os.Stat(filepath.Join(curPath, ".git"))
+				if err != nil {
+					if errors.Is(err, os.ErrNotExist) {
+						err = os.Chdir("..")
+					}
+					cobra.CheckErr(err)
 				} else {
-					fmt.Printf("Git repository initialized.\n")
+					foundGit = true
+					break
 				}
+
+				curWD, err = os.Getwd()
+				cobra.CheckErr(err)
+			}
+
+			// Move back to the app directory.
+			cobra.CheckErr(os.Chdir(prevWD))
+
+			gitPath := "."
+			if !foundGit {
+				// Initialize a new git repository in the app directory.
+				_, err = os.Stat(filepath.Join(gitPath, ".git"))
+				if err != nil && errors.Is(err, os.ErrNotExist) {
+					// Git directory doesn't exist, try to initialize.
+					gitInitCmd := exec.Command("git", "init")
+					if err = gitInitCmd.Run(); err != nil {
+						fmt.Printf("Git repository not initialized: %s.\n", err)
+					} else {
+						fmt.Printf("Git repository initialized.\n")
+					}
+				}
+			} else {
+				// Git repository already exists.
+				gitPath = curPath
 			}
 
-			// Initialize .gitignore.
+			// Initialize .gitignore in the appropriate path.
```

Comment:
It's fine if `.gitignore` lives in the path where the ROFL app is and not the root.

Reply:
OK, I've updated the code, please have another look :)